### PR TITLE
feat: added basic Modal primitive to Gamut

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Added basic Modal primitive to Gamut

This is a very simple version of a modal that acts as a controlled component, receiving isOpen as a prop. It's intentionally much simpler than the overused Modal components in the monolith.

Does not yet include animations. I just want to get this out quickly to unblock certificates & any other work that would use it.

Recording of the modal being opened and closed with keyboard and mouse

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.2-alpha.c42c3f.3+c42c3f31<br/>@codecademy/gamut-templates@2.2.36-alpha.c42c3f.8+c42c3f31<br/>@codecademy/gamut@2.4.12-alpha.c42c3f.8+c42c3f31<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.2-alpha.0114a7.4+0114a709<br/>@codecademy/gamut-templates@2.2.36-alpha.0114a7.9+0114a709<br/>@codecademy/gamut@2.4.12-alpha.0114a7.9+0114a709<br/></pre>

File: packages/gamut/src/Modal/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+.container {
+  bottom: 0;
+  left: 0;
+  position: fixed;
+  right: 0;
+  top: 0;
+}
+
+.container {
```
Comment (jcenglish) [inline packages/gamut/src/Modal/index.module.scss]: are these two `.container` classes a typo?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Modal/index.module.scss]: Ha, yeah. I used to have a `<button />` that received an `onOutsideClick?: () => void` but later removed it for being unnecessary...

File: packages/gamut/src/Modal/__tests__/Modal-test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import Modal from '..';
+
+describe('Icon', () => {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Modal/__tests__/Modal-test.tsx]:
```suggestion
describe('Modal', () => {
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.2-alpha.a6df8c.6+a6df8c71<br/>@codecademy/gamut-templates@2.2.36-alpha.a6df8c.11+a6df8c71<br/>@codecademy/gamut@2.4.12-alpha.a6df8c.11+a6df8c71<br/></pre>

File: packages/gamut/src/Modal/__tests__/Modal-test.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    );
+
+    expect(wrapper.isEmptyRender()).toBe(true);
+  });
+
+  it('renders children when isOpen is true', () => {
+    const children = 'Howdy!';
+    const wrapper = mount(
+      <Modal isOpen>
+        {children}
```
Comment (kipprice) [inline packages/gamut/src/Modal/__tests__/Modal-test.tsx]: why the `{children}` here and  straight `Howdy!` above?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/Modal/__tests__/Modal-test.tsx]: Ha, it's a lines-of-code optimization. The first test never uses `children` so I kept it inline.

File: packages/gamut/src/Modal/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
+@import "~@codecademy/gamut-styles/utils";
+
+.container {
+  align-items: center;
+  background: rgba($color-black, 0.85);
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  left: 0;
```
Comment (kipprice) [inline packages/gamut/src/Modal/styles.module.scss]: nit: is it possible to not have these be alphabetical? the separation between the `left, bottom, ...` is jarring

File: packages/gamut/src/Modal/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  position: fixed;
+  right: 0;
+  top: 0;
+}
+
+// Todo: are these correct?
+.inside {
+  background: $color-gray-100;
+  border-radius: 5px;
```
Comment (jakemhiller) [inline packages/gamut/src/Modal/styles.module.scss]: I wonder if we could just leave off the default styles? maybe just leave out the "inner" div entirely? Most of our Modal styles could use this without passing in any options if it didn't have these default styles

File: packages/gamut/src/Modal/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+@import "~@codecademy/gamut-styles/utils";
+
+.container {
+  align-items: center;
+  background: rgba($color-black, 0.85);
```
Comment (jakemhiller) [inline packages/gamut/src/Modal/styles.module.scss]: ditto here re: style defaults, I think it should either be closer to our current default or just remove it and let the consumer set the background with the container class name.

File: packages/gamut/src/Modal/styles.module.scss (hunk 5 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Modal/styles.module.scss]: Ha, this reminds me of the big Modal discussion in the Gamut sync yesterday. Documenting & summarizing my perspective here: I think the main difference is that I was seeing this Modal _including_ styles as a visual primitize, higher than the base concept of a modal. It's reasonable to move those styles to the (next) DismissableModal template for now.

File: packages/gamut/src/BodyPortal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+export const BodyPortal: React.FC = ({ children }) => {
```
Comment (codecaaron) [inline packages/gamut/src/BodyPortal/index.tsx]: I think our current monolith implementation of this is lets you specific onMount / disMount callbacks. 

Any plans to add this here?

Comment (JoshuaKGoldberg) [reply packages/gamut/src/BodyPortal/index.tsx]: None so far. I'd like to not add features until they're needed; IMO it's debatable whether we'd actually want that.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.2-alpha.d20ed4.8+d20ed425<br/>@codecademy/gamut-templates@2.2.36-alpha.d20ed4.13+d20ed425<br/>@codecademy/gamut@2.4.12-alpha.d20ed4.13+d20ed425<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.2-alpha.f4f9db.9+f4f9db49<br/>@codecademy/gamut-templates@2.2.36-alpha.f4f9db.14+f4f9db49<br/>@codecademy/gamut@2.4.12-alpha.f4f9db.14+f4f9db49<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.3.5-alpha.27f01b.7+27f01b23<br/>@codecademy/gamut-templates@2.3.3-alpha.27f01b.7+27f01b23<br/>@codecademy/gamut@2.6.2-alpha.27f01b.7+27f01b23<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/enzyme-helpers@1.1.9-alpha.b22c27.99+b22c27fb<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.1-alpha.e83ec1.10+e83ec15a<br/></pre>

**Comment by JoshuaKGoldberg** [comment]
Ugh, go away codecov...

**Pull request merged**